themes/default/content/blog/pulumi-go-boilerplate-v2/index.md

```diff
@@ -0,0 +1,218 @@
+---
+title: The Easier Way to Create Pulumi Providers in Go
+allow_long_title: true
+date: 2023-01-03
+authors: 
+  - kyle-dixler
+tags:
+  - go
+  - packages
+meta_desc: A major update to the Pulumi Provider Boilerplate simplifies the provider development process.
+meta_image: meta.png
+---
+
+A major update to the Pulumi Provider Boilerplate simplifies the provider development process.
+
+<!--more-->
+
+## Background
+
+The Pulumi Provider Boilerplate [(view on GitHub)](https://github.com/pulumi/pulumi-provider-boilerplate) is a template repository
+that contains a minimal example of a working provider and has served as a starting point for writing your own provider.
+
+For those who may not be familiar with provider authoring, it involves two distinct parts:
+
+1. Provider implementation
+2. SDK generation
+
+Providers are implemented as gRPC servers that handle requests to perform CRUD (create, read, update, delete) operations on resources on behalf of the Pulumi engine.
+SDK generation, on the other hand, involves generating source code libraries for a Pulumi program to utilize the provided resources.
+
+In the past, implementing a provider using the old provider boilerplate could be difficult because you had to implement:
+
+- your provider's resources
+- the gRPC server
+- SDK code generation to consume these resources in a Pulumi program
+
+## New Provider Boilerplate
+
+We are excited to announce that we've updated the Pulumi Provider Boilerplate to make this easier!
+
+**Note:** This update will not have any impact on existing providers that are utilizing an older version of the repository.
+
+This major update brings a wealth of usability improvements to the Pulumi Provider Boilerplate by incorporating our brand
+new Pulumi Go Provider SDK [(view on GitHub)](https://github.com/pulumi/pulumi-go-provider).
+Now, you can implement your provider in Go by defining the provider's metadata, resources, and functions, and then building the provider binary.
```

Comment:
Would like to work in the term "code-first" here, elaborating that we've replaced the need for providers to declare a separate `schema.json` for SDK codegen.